tds.go

```diff
@@ -1145,6 +1148,25 @@ initiate_connection:
 		passthrough := passthroughConn{c: &handshakeConn}
 		tlsConn := tls.Client(&passthrough, &config)
 		err = tlsConn.Handshake()
+		if !p.trustServerCertificate && // not forcibly trusting server certificate
+			p.disableVerifyHostname && // carry out manual verification
+			err == nil { // no handshake error
+
+			certs := tlsConn.ConnectionState().PeerCertificates
+			opts := x509.VerifyOptions{
+				DNSName:       tlsConn.ConnectionState().ServerName,
+				Intermediates: x509.NewCertPool(),
+				Roots:         config.RootCAs,
+			}
+			for i, cert := range certs {
+				if i == 0 {
+					continue
+				}
+				opts.Intermediates.AddCert(cert)
+			}
+			_, err = certs[0].Verify(opts)
+		}
+
```

Comment:
It looks like tlsConn.Handshake() can return an error before we get to line 1151. We should have error handling there as well as at the end of this `if` block

Reply:
There's no need. This block of conditional code is only executed if err is nil, plus the other conditions.